### PR TITLE
Use the 'Application Support' directory to store config files on OS X

### PR DESCRIPTION
**Pull request by AaronVanGeffen**

### CODE REVIEW

**Comment by LRFLEW** [comment]
I should look at the existing PR's before I make one next time. :P I made #2465 without realizing you made this. However, we went about it two different ways, and I don't know which one is better. I'll leave my PR open for now, and close it if yours gets merged.

File: src/platform/posix.c [hunk 1]
```diff
@@ -567,7 +567,16 @@ void platform_resolve_user_data_path()
 
 		strncat(buffer, homedir, MAX_PATH - 1);
 		strncat(buffer, separator, MAX_PATH - strnlen(buffer, MAX_PATH) - 1);
+
+		// Use the 'Application Support' directory to store config files on OS X.
+#if defined(__APPLE__) && defined(__MACH__)
+		strncat(buffer, "Library", MAX_PATH - strnlen(buffer, MAX_PATH) - 1);
+		strncat(buffer, separator, MAX_PATH - strnlen(buffer, MAX_PATH) - 1);
+		strncat(buffer, "Application Support", MAX_PATH - strnlen(buffer, MAX_PATH) - 1);
+#elif
```
Comment (LRFLEW) [inline src/platform/posix.c]: This should be `#else`

Comment (AaronVanGeffen) [inline src/platform/posix.c]: Oops, thanks!

**Comment by LRFLEW** [comment]
The main reason I think my version is better is not the separation of code in the `linux.c` and `osx.c` files, but that my version ignores the `$XDG_CONFIG_HOME` variable on OS X because its use doesn't really make sense there (though that difference is relatively minor).

**Pull request closed without merging**